docs/SETUP/BACK_END.md

```diff
@@ -2,55 +2,72 @@
+The project needs certain specific and sensitive pieces of information to run. We store these in the user's development environment and not in source control.
 
-For Mac and Linux, you can permanently store the environmental variables in your command line shell's configuration file, e.g. `~/.bashrc`, `~/.bash_profile`, `~/.zshrc` or `~/.profile`.  Add a line `export VAR_NAME=VALUE` in your file and run `source file` to read it in when newly created.  Any new shells will automatically have the new environment.
+#### Using a .env File
 
-For Windows, you can set environmental variables under System -> Advanced or you can download a terminal emulator such as [Git Bash](https://gitforwindows.org/) and follow the instructions for Mac and Linux above.  A terminal emulator is reccommended.
+1. Create a file named `.env` in the `/data` subdirectory of your project.
+2. Add the following environment variables to the `.env` file:
 
-All of your local environmental variables will be passed through to docker-compose so if you have them locally you should not have to hard-code them as indicated below.
-
-#### PostgreSQL
+```sh
+POSTGRES_PASSWORD=a-strong-password-here
+VACANT_LOTS_DB=postgresql://postgres:${POSTGRES_PASSWORD}@localhost/vacantlotdb
+```
 
-Create an environmental variable called `POSTGRES_PASSWORD` and set its value to a new, strong password to use for your local postgres installation in Docker.  After that, add the below variable to configure the full postgres connection string:
-`VACANT_LOTS_DB="postgresql://postgres:${POSTGRES_PASSWORD}@localhost/vacantlotdb"`
+All local environmental variables will be passed through to docker-compose, so if you have them set up in the `.env` file, you should not need to hard-code them elsewhere.
 
 ### Docker Build
 
 Docker is a platform that allows you to containerize and run applications in isolated environments, making it easier to manage dependencies and ensure consistent deployments. Download the [latest version of Docker Desktop for your operating system](https://www.docker.com/products/docker-desktop/).
 
-We use [docker-compose](https://docs.docker.com/compose/) to manage the backend Docker services.  The `data/docker-compose.yaml` file defines the services.  The only service that runs perpetually in Docker is `postgres`.  The other services are one-time batch jobs to build the data sets.
+We use [docker-compose](https://docs.docker.com/compose/) to manage the backend Docker services. The `data/docker-compose.yaml` file defines the services. The only service that runs perpetually in Docker is `postgres`. The other services are one-time batch jobs to build the data sets.
+
+1. The first time you set up your backend, or any time either of the two Docker files change, build the Docker services by running:
+
+   ```sh
+   docker-compose build
+   ```
+
+2. When this has finished, also run:
+
+   ```sh
+   docker-compose build postgres
```

Comment:
`docker compose build` should build all the containers, including `postgres` but I understand your experience was different so I guess we can leave this in.